Add Recommended Movies Screen

Mostafa-khaled21 opened the pull request:
https://github.com/user-attachments/assets/94dfaa66-03fe-49c5-9e9f-4f08a6d979e4

Mostafa-khaled21 committed:
feat: Add Recommended Movies Screen

HendSayed25 reviewed (dismissed):
Great Mostafa !

AhmedMohamed056 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/MovieDetailsInteractionListener.kt:
```diff
@@ -30,4 +30,6 @@ interface MovieDetailsInteractionListener {
 
+    fun navigateToMovieRecommendation(movieId: Int, title: String)
+
 }
```
You didn't use it

Mostafa-khaled21 replied:
i use it in movie details screen 
                   interaction.navigateToMovieRecommendation(state.movie.id, state.movie.title)
line 191

K-M-9 commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/RecommendedMoviesViewModel.kt:
```diff
@@ -0,0 +1,42 @@
+
+class RecommendedMoviesViewModel(
+    val title: String = "",
+    private val movieId: Int,
```
Use the `SavedStateHandle` within your `ViewModel` instead of passing values through its constructor.

K-M-9 commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/TransitionLazyColumnToGridMovie.kt:
```diff
@@ -0,0 +1,127 @@
+                previousIndex = index
+                previousScrollOffset = offset
+            }
+    }
```
You can move this repeated code into a `separate` function and reuse it to reduce `duplication`.

K-M-9 commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/TransitionLazyColumnToGridMovie.kt:
```diff
@@ -0,0 +1,127 @@
+            }
+        }
+    }
+}
```
I think `LazyColumn` and `LazyVerticalGrid` should each be placed in their own `function`.

K-M-9 commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/RecomendedContent.kt:
```diff
@@ -0,0 +1,110 @@
+@Composable
+fun RecommendedContent(
+    title: String,
+    navController: NavController,
```
Instead of passing the `NavController` to the `screen`, use a `callback` function to handle navigation in `Navigation` file .

Mostafa-khaled21 replied:
please check again

K-M-9 commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/TransitionLazyColumnToGridMovie.kt:
I noticed that you’re not `handling` the `Loading` and `Error` states in your `Paging` `UI` implementation.

Mostafa-khaled21 replied:
check RecommendedMovieScreen now

Mostafa-khaled21 committed:
Refactor: Extract scroll direction observation and list/grid views

Mostafa-khaled21 committed:
Refactor RecommendedMoviesViewModel to use SavedStateHandle

Mostafa-khaled21 committed:
Refactor RecommendedMovieScreen to improve error handling and simplify ViewModel injection

Mostafa-khaled21 committed:
Refactor RecommendedMovieScreen and RecommendedContent

- Move navigation logic from RecommendedContent to RecommendedMovieScreen
- Add onItemClick and onBackClick parameters to RecommendedContent

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/develop' into fix/Fix-Bugs-in-Recommendations-Screen-movie

# Conflicts:
#	app/src/main/java/com/giraffe/cineverseapp/di/viewModelModule.kt
#	presentation/details/src/main/java/com/giraffe/details/navigation/DetailsApiImp.kt
#	presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screen/MovieDetailsScreen.kt

Mostafa-khaled21 committed:
Refactor: Update recommended movies navigation and resolve conflict

HendSayed25 commented on presentation/details/src/main/java/com/giraffe/details/nav/DetailsNavGraph.kt:
```diff
@@ -54,6 +56,18 @@ fun DetailsNavGraph(
+        recommendedMoviesRoute(
+            onBackClick = {
+                val backed = navController.navigateUp()
+                if (!backed) {
```
what you mean in this section ?
why apply check in back variable and what's the back() ?

skaik-mo commented on app/src/main/java/com/giraffe/cineverseapp/di/viewModelModule.kt:
```diff
@@ -20,4 +25,13 @@ val viewModelModule = module {
     viewModelOf(::SearchResultViewModel)
     //viewModelOf(::RecommendedSeriesViewModel)
     //viewModel { (seriesId: Long) -> SeriesRecommendationViewModel(get(), seriesId) }
```
Remove this comment to keep the code clean.

HendSayed25 commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/RecommendedMovieRoute.kt:
```diff
@@ -1,8 +1,32 @@
+    onMovieClick: (Int) -> Unit
+) {
+    composable(
+        route = "recommended_movies/{movieId}/{title}",
```
it's better to making name of route be const and the args :

const val RECOMMENDED_MOVIE_ROUTE = "recommended_movies"
const val MOVIE_ID_ARG = "movieId"
const val TITLE_ARG = "title"

skaik-mo replied:
It’s better to define a `@Serializable` data class for the route. This improves type safety, makes navigation arguments easier to manage, and keeps your navigation code cleaner and more maintainable.

HendSayed25 reviewed (dismissed):
Good job

skaik-mo commented on presentation/details/src/main/java/com/giraffe/details/models/MovieUi.kt:
```diff
@@ -10,9 +11,24 @@ data class MovieUi(
-)
+) {
+    companion object {
+        fun fromEntity(movie: Movie): MovieUi = MovieUi(
```
Why use `fromEntity` if you're already using `MovieUi()` on line 45? Also, rename `MovieUi` to start with a lowercase letter to match Kotlin conventions.

skaik-mo replied:
you forget rename `MovieUi` to start with a lowercase letter

skaik-mo commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/RecomendedContent.kt:
```diff
@@ -0,0 +1,108 @@
+        LazyColumn(
+            modifier = modifier
+                .background(Theme.color.background.screen)
+                .statusBarsPadding()
```
`statusBarsPadding` is redundant here, `systemBarsPadding` already handles both top and bottom system bars.

skaik-mo commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/RecomendedContent.kt:
```diff
@@ -0,0 +1,108 @@
+            }
+        }
+
+        val layoutDirection = LocalLayoutDirection.current
```
System alignment is handled by default, no need to set it manually.

skaik-mo commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/RecommendedMovieScreen.kt:
```diff
@@ -0,0 +1,119 @@
+            textAlign = TextAlign.Center
+        )
+
+        Spacer(modifier = Modifier.height(16.dp))
```
Use `verticalArrangement` instead of multiple Spacers for cleaner and more efficient layout.

skaik-mo commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/RecommendedMovieScreen.kt:
```diff
@@ -0,0 +1,119 @@
+            )
+        ) {
+            Text(
+                text = "Retry",
```
Use stringResource(...) instead of hardcoded text to enable localization.

skaik-mo commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/RecommendedMoviesViewModel.kt:
```diff
@@ -0,0 +1,45 @@
+    }
+        .flow
+        .cachedIn(viewModelScope)
+        .map { pagingData ->
```
Extract the mapping logic into a separate function to improve clarity and readability.

skaik-mo reviewed (changes requested):
### There’s extra padding applied here, please double-check and adjust it to match the design specifications.

<img width="349" height="735" alt="image" src="https://github.com/user-attachments/assets/608744ab-a160-47b6-b24c-3ecd2aa26a61" />

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/develop' into fix/Fix-Bugs-in-Recommendations-Screen-movie

Mostafa-khaled21 committed:
Refactor: Use RecommendedMovieArgs for navigation

Mostafa-khaled21 committed:
Refactor: Simplify MovieUi creation and remove unused code

Mostafa-khaled21 committed:
Refactor: Improve recommended movies screen UI and navigatio

skaik-mo commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/RecommendedMovieRoute.kt:
```diff
@@ -0,0 +1,34 @@
+import androidx.navigation.navArgument
+import com.giraffe.details.screens.recommended.movie.utils.RecommendedMovieArgs
+
+fun NavController.navigateToRecommendedMoviesScreen(args: RecommendedMovieArgs) {
```
Pass `movieId` and `title` directly, then use `RecommendedMovieArgs(movieId, title)` for cleaner code.

skaik-mo commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/utils/RecommendedMovieNavArgs.kt:
```diff
@@ -0,0 +1,23 @@
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class RecommendedMovieArgs(
```
Use a simple @Serializable data class like this:
```
@Serializable  
data class RecommendedMovieArgs(  
    val movieId: Int,  
    val title: String  
)
```
It keeps things clean and easy to manage when passing arguments.

skaik-mo commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/utils/RecommendedMovieNavArgs.kt:
```diff
@@ -0,0 +1,23 @@
+package com.giraffe.details.screens.recommended.movie.utils
+
+import android.net.Uri
```
Move the file alongside its screen for better project structure.

skaik-mo commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/RecommendedMovieRoute.kt:
```diff
@@ -0,0 +1,34 @@
+import com.giraffe.details.screens.recommended.movie.utils.RecommendedMovieArgs
+
+fun NavController.navigateToRecommendedMoviesScreen(args: RecommendedMovieArgs) {
+    navigate(args.toRoute())
```
Use `RecommendedMovieArgs(movieId, title)` instead of `args.toRoute() `for simplicity and clarity.

skaik-mo commented on presentation/details/src/main/java/com/giraffe/details/screens/recommended/movie/RecommendedMovieRoute.kt:
```diff
@@ -0,0 +1,34 @@
+    onMovieClick: (Int) -> Unit
+) {
+    composable(
+        route = RecommendedMovieArgs.fullRoute,
```
Pass the data class directly in the composable like this: `composable<RecommendedMovieArgs>{ }`

skaik-mo replied:
Inside the composable, you can access `movieId` and `title` using `it.toRoute<RecommendedMovieArgs>()`, then pass them directly to the ViewModel

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/develop' into fix/Fix-Bugs-in-Recommendations-Screen-movie

# Conflicts:
#	app/src/main/java/com/giraffe/cineverseapp/di/viewModelModule.kt

Mostafa-khaled21 committed:
Refactor: Use type-safe navigation for recommended movies

Mostafa-khaled21 committed:
Merge remote-tracking branch 'origin/develop' into fix/Fix-Bugs-in-Recommendations-Screen-movie

# Conflicts:
#	app/src/main/java/com/giraffe/cineverseapp/di/viewModelModule.kt
#	presentation/details/src/main/java/com/giraffe/details/nav/DetailsNavGraph.kt

Mostafa-khaled21 committed:
Refactor: Recommended movie screen

- Created `RecommendedEffectMovie` for navigation.
- Updated `RecommendedMoviesViewModel` to implement `ViewModel` and handle navigation effects.
- Modified `RecommendedContent` and `TransitionLazyColumnToGridMovie` to use `RecommendedInteractionListener`.
- Adjusted `DetailsNavGraph` to use the updated navigation logic.
- Removed unused `SeriesRecommendationViewModel` from `viewModelModule`.
- Added `RecommendedInteractionListener` interface.

Mostafa-khaled21 committed:
Refactor: Rename `MovieUi` extension function to `toMovieUi`
